R4.2: restoring backups crashes

## Observation

openQA test in scenario qubesos-4.2-pull-requests-x86_64-system_tests_backup@64bit fails in
[TC_00_Backup](https://openqa.qubes-os.org/tests/60361/modules/TC_00_Backup/steps/1)

```
Traceback (most recent call last):
  File "/usr/lib/python3.11/site-packages/qubes/tests/integ/backup.py", line 223, in restore_backup
    restore_op = self.loop.run_until_complete(
                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib64/python3.11/asyncio/base_events.py", line 653, in run_until_complete
    return future.result()
           ^^^^^^^^^^^^^^^
  File "/usr/lib64/python3.11/concurrent/futures/thread.py", line 58, in run
    result = self.fn(*self.args, **self.kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.11/site-packages/qubesadmin/backup/restore.py", line 967, in __init__
    self.backup_app = self._process_qubes_xml()
                      ^^^^^^^^^^^^^^^^^^^^^^^^^
qubesadmin.exc.QubesException: unable to extract the qubes backup. Check extracting process errors.

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/lib64/python3.11/contextlib.py", line 81, in inner
    return func(*args, **kwds)
           ^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.11/site-packages/qubes/tests/integ/backup.py", line 369, in test_000_basic_backup
    self.restore_backup()
  File "/usr/lib/python3.11/site-packages/qubes/tests/integ/backup.py", line 222, in restore_backup
    with self.assertNotRaises(qubesadmin.exc.QubesException):
  File "/usr/lib/python3.11/site-packages/qubes/tests/__init__.py", line 281, in __exit__
    raise self.failureException(
AssertionError: QubesException('unable to extract the qubes backup. Check extracting process errors.') raised, traceback:
  File "/usr/lib/python3.11/site-packages/qubes/tests/integ/backup.py", line 223, in restore_backup
    restore_op = self.loop.run_until_complete(
                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib64/python3.11/asyncio/base_events.py", line 653, in run_until_complete
    return future.result()
           ^^^^^^^^^^^^^^^
  File "/usr/lib64/python3.11/concurrent/futures/thread.py", line 58, in run
    result = self.fn(*self.args, **self.kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.11/site-packages/qubesadmin/backup/restore.py", line 967, in __init__
    self.backup_app = self._process_qubes_xml()
                      ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.11/site-packages/qubesadmin/backup/restore.py", line 1378, in _process_qubes_xml
    raise QubesException(
```

## Test suite description

1. Make a backup
2. Restore it

See test code at https://github.com/QubesOS/qubes-core-admin/blob/main/qubes/tests/integ/backup.py#L360

## Reproducible

Fails since (at least) Build [2023021319-4.2](https://openqa.qubes-os.org/tests/60229)


## Expected result

Last good: [2023011805-4.2](https://openqa.qubes-os.org/tests/58292) (or more recent)


## Further details

Always latest result in this scenario: [latest](https://openqa.qubes-os.org/tests/latest?arch=x86_64&distri=qubesos&flavor=pull-requests&machine=64bit&test=system_tests_backup&version=4.2)

--- a/qubesadmin/backup/restore.py
+++ b/qubesadmin/backup/restore.py
@@ -38,6 +38,8 @@ import sys
 import tempfile
 import termios
 import time
+# only for a python bug workaround
+import concurrent.futures.thread
 
 import collections
 
@@ -300,28 +302,16 @@ def launch_scrypt(action, input_name, output_name, passphrase):
     p.pty = pty
     return p
 
-def _fix_logging_lock_after_fork():
+def _fix_threading_after_fork():
     """
-    HACK:
-    This is running in a child process, parent might hold some lock
-    while fork was called (but will be released only in a parent
-    process). This specifically applies to a logging module and
-    results in a deadlock (if one is unlucky). "Fix" this by
-    reinitialize a lock on all registered logging handlers
-    just after a fork() call, until fixed upstream:
+    HACK
+    Clear thread queues after fork (threads are gone at this point),
+    otherwise atexit callback will crash.
 
-    https://bugs.python.org/issue6721
+    https://github.com/python/cpython/issues/88110
     """
-    if not hasattr(logging, '_handlerList'):
-        return
-
     # pylint: disable=protected-access
-    for handler_ref in logging._handlerList:
-        handler = handler_ref()
-        if handler is None:
-            continue
-        if handler.lock:
-            handler.lock = type(handler.lock)()
+    concurrent.futures.thread._threads_queues.clear()
 
 
 class ExtractWorker3(Process):
@@ -437,7 +427,7 @@ class ExtractWorker3(Process):
 
     def run(self):
         try:
-            _fix_logging_lock_after_fork()
+            _fix_threading_after_fork()
             self.__run__()
         except Exception:
             # Cleanup children

